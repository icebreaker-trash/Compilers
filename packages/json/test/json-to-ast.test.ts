// @ts-ignore
const parse = require('json-to-ast')
describe('json-to-ast', () => {
  it('Common Usage', () => {
    const settings = {
      // Appends location information. Default is <true>
      loc: true,
      // Appends source information to nodeâ€™s location. Default is <null>
      source: 'data.json'
    }
    // @ts-ignore
    const ast = parse('{"a": 1}', settings)
    expect(ast).toMatchSnapshot()
  })
})
