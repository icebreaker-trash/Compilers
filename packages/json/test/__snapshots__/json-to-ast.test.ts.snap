// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json-to-ast Common Usage 1`] = `
{
  "children": [
    {
      "key": {
        "loc": {
          "end": {
            "column": 5,
            "line": 1,
            "offset": 4,
          },
          "source": "data.json",
          "start": {
            "column": 2,
            "line": 1,
            "offset": 1,
          },
        },
        "raw": ""a"",
        "type": "Identifier",
        "value": "a",
      },
      "loc": {
        "end": {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
        "source": "data.json",
        "start": {
          "column": 2,
          "line": 1,
          "offset": 1,
        },
      },
      "type": "Property",
      "value": {
        "loc": {
          "end": {
            "column": 8,
            "line": 1,
            "offset": 7,
          },
          "source": "data.json",
          "start": {
            "column": 7,
            "line": 1,
            "offset": 6,
          },
        },
        "raw": "1",
        "type": "Literal",
        "value": 1,
      },
    },
  ],
  "loc": {
    "end": {
      "column": 9,
      "line": 1,
      "offset": 8,
    },
    "source": "data.json",
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "Object",
}
`;
